.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,27 @@
+name: Build and Publish
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v7
+        with:
+          java-version: graalvm@20.0.0
+      - name: Check formatting
+        run: sbt scalafmtCheckAll
+      - name: Compile
+        run: sbt Compile/compile Test/compile
+      - name: Test
+        run: sbt "set ThisBuild/coverageEnabled := true" test
+      - uses: codecov/codecov-action@v1
+      - name: Publish
+        env:
+          ARTIFACTORY_USERNAME: ${{ secrets.ArtifactoryUsername }}
+          ARTIFACTORY_PASSWORD: ${{ secrets.ArtifactoryPassword }}
+        run: sbt publish
```

Comment:
I don't think we'll be able to test this one without merging to `master` and pushing a test tag.